src/dev/flang/ast/Feature.java

```diff
@@ -1591,6 +1582,22 @@ private void checkChoiceAndAddInternalFields(Resolution res)
         }
       }
 
+    res._module.forEachDeclaredOrInheritedFeature(this,
+                                                  p ->
+      {
+        // choice type must not have any fields
+        if (p.isField() && !p.isOuterRef() &&
+            (!Errors.any() || !(p instanceof Feature pf && (pf.isArtificialField() || /* do not report auto-generated fields like `result` in choice if there are other problems */
```

Comment:
This is hard to read for me. I think we can apply De Morgan here?:
`!(Errors.any() && (p instanceof Feature pf && (pf.isArtificialField() || pf.isResultField()))`